app/src/main/java/zapsolutions/zap/transactionHistory/transactionDetails/InvoiceDetailBSDFragment.java

```diff
@@ -0,0 +1,221 @@
+package zapsolutions.zap.transactionHistory.transactionDetails;
+
+import android.graphics.Bitmap;
+import android.os.Bundle;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageButton;
+import android.widget.ImageView;
+import android.widget.TextView;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import com.github.lightningnetwork.lnd.lnrpc.Invoice;
+import com.google.android.material.bottomsheet.BottomSheetDialogFragment;
+import com.google.protobuf.ByteString;
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+import net.glxn.qrgen.android.QRCode;
+import zapsolutions.zap.R;
+import zapsolutions.zap.util.MonetaryUtil;
+import zapsolutions.zap.util.Wallet;
+import zapsolutions.zap.util.ZapLog;
+
+import java.text.DateFormat;
+import java.util.Date;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class InvoiceDetailBSDFragment extends BottomSheetDialogFragment {
+
+    public static final String TAG = InvoiceDetailBSDFragment.class.getName();
+    public static final String ARGS_TRANSACTION = "TRANSACTION";
+
+    private TextView mTransactionDescription;
+    private TextView mAmountLabel;
+    private TextView mAmount;
+    private TextView mMemoLabel;
+    private TextView mMemo;
+    private TextView mDateLabel;
+    private TextView mDate;
+    private TextView mExpiryLabel;
+    private TextView mExpiry;
+    private ImageView mQRCodeView;
+
+    @Nullable
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.bsd_invoice_detail, container);
+
+        mTransactionDescription = view.findViewById(R.id.transactionDescription);
+        mAmountLabel = view.findViewById(R.id.amountLabel);
+        mAmount = view.findViewById(R.id.amount);
+        mMemoLabel = view.findViewById(R.id.memoLabel);
+        mMemo = view.findViewById(R.id.memo);
+        mDateLabel = view.findViewById(R.id.dateLabel);
+        mDate = view.findViewById(R.id.date);
+        mExpiryLabel = view.findViewById(R.id.expiryLabel);
+        mExpiry = view.findViewById(R.id.expiry);
+        mQRCodeView = view.findViewById(R.id.requestQRCode);
+
+        ImageButton closeButton = view.findViewById(R.id.closeButton);
+        closeButton.setOnClickListener(view1 -> dismiss());
+
+        if (getArguments() != null) {
+            ByteString transactionString = (ByteString) getArguments().getSerializable(ARGS_TRANSACTION);
+            try {
+                bindInvoice(transactionString);
+            } catch (InvalidProtocolBufferException exception) {
```

Comment:
Could be simplified to `} catch (InvalidProtocolBufferException | NullPointerException exception) {`